Bisal/assemblycodes/sort.s

```diff
@@ -0,0 +1,61 @@
+#  ============================================================================
+#  Filename:    sort.s 
+#  Description: File consists of code based for insertion sort algorithm
+#  Author:      Bisal Saeed
+#  Date:        7/12/2024
+#  ============================================================================
+
+.section .text
+.globl _start
+
+_start:
+    .data
+    .align 2
+array:
+    .word 5, 2, 4, 6, 1, 3    # Array to be sorted
+
+array_size = (. - array) / 4  # Calculate array size in words
+
+    # Initialization
+    la   t0, array       # address of array 
+    li   t1, 1           # counter variable
+
+outer_loop:
+    # exit if t1 > = array size 
+    bge  t1, array_size, end_sort  
+
+    # Inner loop
+    mv   t2, t1           # t2 = i (index of element to be inserted)
+    lw   t3, 0(t0)        # Load array[i] into t3
+    li   t4, 0            # Initialize (index for comparison)
+
+inner_loop:
+    # Exit loop if j >= i 
+    blt  t4, t2, check_next  

```

Comment:
Where is this "check_next" label? It is not in this file so this line will generate an error. You are supposed to commit a working and error free code.